File: jdroid-android-core/src/main/java/com/jdroid/android/share/AppSharingItem.java
package com.jdroid.android.share;

import android.app.Activity;
import android.support.annotation.NonNull;

public abstract class AppSharingItem extends SharingItem {

	private SharingData sharingData;

	public AppSharingItem(@NonNull SharingData sharingData) {
		this.sharingData = sharingData;
	}

	@Override
	public void share(Activity activity) {
		SharingDataItem sharingDataItem = sharingData.getShareInfoItemMap().get(getSharingMedium().getName());
		if (sharingDataItem == null) {
			sharingDataItem = sharingData.getDefaultSharingDataItem();
		}

		String text = sharingDataItem.getText() != null ? sharingDataItem.getText() : sharingData.getDefaultSharingDataItem().getText();
		String link = sharingDataItem.getLink() != null ? sharingDataItem.getLink() : sharingData.getDefaultSharingDataItem().getLink();
		if (link != null) {
			text = text.replace("${link}", link);
		}
		ShareUtils.share(activity, getSharingMedium(), sharingData.getShareKey(), text);
	}

	@NonNull
	public abstract SharingMedium getSharingMedium();

	@Override
	public String getApplicationId() {
		return getSharingMedium().getApplicationId();
	}

}

File: jdroid-android-core/src/main/java/com/jdroid/android/share/AppSharingItem.kt
package com.jdroid.android.share

import android.app.Activity

abstract class AppSharingItem(private val sharingData: SharingData) : SharingItem() {

    abstract fun getSharingMedium(): SharingMedium

    override fun share(activity: Activity) {
        var sharingDataItem = sharingData.shareInfoItemMap[getSharingMedium().getMediumName()]
        if (sharingDataItem == null) {
            sharingDataItem = sharingData.defaultSharingDataItem
        }

        var text = if (sharingDataItem!!.text != null) sharingDataItem.text else sharingData.defaultSharingDataItem.text
        val link = if (sharingDataItem.link != null) sharingDataItem.link else sharingData.defaultSharingDataItem.link
        if (link != null) {
            text = text.replace("\${link}", link)
        }
        ShareUtils.share(activity, getSharingMedium(), sharingData.shareKey, text)
    }

    override fun getApplicationId(): String? {
        return getSharingMedium().applicationId
    }
}
